rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -326,25 +331,38 @@ bool Player::play_next()
 
   // Temporary take over playback from play_messages_from_queue()
   std::lock_guard<std::mutex> main_play_loop_lk(skip_message_in_main_play_loop_mutex_);
-  skip_message_in_main_play_loop_ = true;
+
```

Comment:
It doesn't make sense to move `skip_message_in_main_play_loop_ = true;` inside loop while still holding relevant `skip_message_in_main_play_loop_mutex_` mutex.  
Please return it back.